spring-integration-gemfire/src/test/java/org/springframework/integration/gemfire/metadata/GemfireMetadataStoreCacheListenerTests.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.gemfire.metadata;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.CacheFactory;
+import org.apache.geode.cache.Region;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.springframework.integration.metadata.MetadataStoreListenerAdapter;
+import org.springframework.util.Assert;
+
+/**
+ * @author Venil Noronha
+ * @since 5.0
+ *
+ */
+public class GemfireMetadataStoreCacheListenerTests {
+
+	private static Cache cache;
+
+	private static GemfireMetadataStore metadataStore;
+
+	private static Region<Object, Object> region;
+
+	@BeforeClass
+	public static void startUp() throws Exception {
+		cache = new CacheFactory().create();
+		metadataStore = new GemfireMetadataStore(cache);
+		region = cache.getRegion(GemfireMetadataStore.KEY);
+	}
+
+	@AfterClass
+	public static void cleanUp() {
+		if (region != null) {
+			region.close();
+		}
+		if (cache != null) {
+			cache.close();
+			Assert.isTrue(cache.isClosed(), "Cache did not close after close() call");
+		}
+	}
+
+	@Before
+	@After
+	public void setup() {
+		if (region != null) {
+			region.clear();
+		}
+	}
+
+	@Test
+	public void testAdd() throws InterruptedException {
+		String testKey = "key";
+		String testValue = "value";
+
+		CountDownLatch latch = new CountDownLatch(1);
+		StringBuilder actualKey = new StringBuilder();
```

Comment:
That's a bit unusual approach for testing data to use `StringBuilder`. Typically we rely on the `AtomicReference`.
But let it be.